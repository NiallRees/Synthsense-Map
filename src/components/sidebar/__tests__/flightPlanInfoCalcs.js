import FlightPlanInfoCalcs from '../flightPlanInfoCalcs';
import React from 'react';
var enzyme = require('enzyme');


it('Matches values and snapshot with arbitrary takeoff and sensors', () => {
  const component = enzyme.shallow(
    <FlightPlanInfoCalcs 
      planFlightParameters={{
        'horizontalSpeed':5,
        'ascendingSpeed':2,
        'descendingSpeed':3,
        'altitude': 30
      }}
      planTakeoff={{
        'elevation': 3,
        'latitude': 51.470188,
        'longitude': -0.3064749
      }}
      planRouteMarkers={[
        {'elevation': 6,
        'latitude': 51.4721273,
        'longitude': -0.3043489},
        {'elevation': 4,
        'latitude': 51.4711572,
        'longitude': -0.3022958}
      ]}
    />
  );
  expect(component.find('#bottom-flight-info-value').at(1).text()).toBe('748') // distance
  expect(component.find('#bottom-flight-info-value').at(2).text()).toBe('93') // ascent
  expect(component.find('#bottom-flight-info-value').at(0).text()).toBe('03:47') // time
  expect(component).toMatchSnapshot();
})

describe('Test FlightPlanInfoCalcs', () => {
  it('Renders correctly without props', () => {
    const component = enzyme.shallow(<FlightPlanInfoCalcs/>);
    expect(component).toMatchSnapshot();
  });

  it('Matches snapshot with no takeoff or sensors', () => {
    const component = enzyme.shallow(
      <FlightPlanInfoCalcs 
        planFlightParameters={{
          'horizontalSpeed':0,
          'ascendingSpeed':0,
          'descendingSpeed':0
        }}
        planTakeoff={null}
        planRouteMarkers={[]}
      />
    );
    expect(component.find('#bottom-flight-info-value').at(0).text()).toBe('00:00') // time
    expect(component.find('#bottom-flight-info-value').at(1).text()).toBe('0') // distance
    expect(component.find('#bottom-flight-info-value').at(2).text()).toBe('0') // ascent
    expect(component).toMatchSnapshot();
  })

  it('Matches values with no takeoff or sensors', () => {
    const component = enzyme.shallow(
      <FlightPlanInfoCalcs 
        planFlightParameters={{
          'horizontalSpeed':5,
          'ascendingSpeed':5,
          'descendingSpeed':5,
          'altitude': 30
        }}
        planTakeoff={null}
        planRouteMarkers={[]}
      />
    );
    expect(component.find('#bottom-flight-info-value').at(0).text()).toBe('00:00') // time
    expect(component.find('#bottom-flight-info-value').at(1).text()).toBe('0') // distance
    expect(component.find('#bottom-flight-info-value').at(2).text()).toBe('0') // ascent
    expect(component).toMatchSnapshot();
  })

});
