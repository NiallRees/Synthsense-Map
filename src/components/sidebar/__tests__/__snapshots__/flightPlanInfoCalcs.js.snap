// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FlightPlanInfoCalcs Matches snapshot with no takeoff or sensors 1`] = `ShallowWrapper {}`;

exports[`Test FlightPlanInfoCalcs Matches values and snapshot with arbitrary takeoff and sensors 1`] = `ShallowWrapper {}`;

exports[`Test FlightPlanInfoCalcs Matches values with no takeoff or sensors 1`] = `ShallowWrapper {}`;

exports[`Test FlightPlanInfoCalcs Renders correctly without props 1`] = `ShallowWrapper {}`;
